swap/swap.go

```diff
@@ -120,74 +119,97 @@ func swapRotatingFileHandler(logdir string) (log.Handler, error) {
+	// verify that backendURL is not empty
 	if backendURL == "" {
-		return nil, errors.New("swap init error: no backend URL given")
+		return nil, errors.New("no backend URL given")
 	}
-	log.Info("connecting to SWAP API", "url", backendURL)
+	auditLog.Info("connecting to SWAP API", "url", backendURL)
 	// initialize the balances store
-	stateStore, err := state.NewDBStore(filepath.Join(dbPath, "swap.db"))
-	if err != nil {
-		return nil, fmt.Errorf("swap init error: %s", err)
+	var stateStore state.Store
+	if stateStore, err = state.NewDBStore(filepath.Join(dbPath, "swap.db")); err != nil {
+		return nil, fmt.Errorf("error while initializing statestore: %v", err)
 	}
 	if params.DisconnectThreshold <= params.PaymentThreshold {
-		return nil, fmt.Errorf("swap init error: disconnect threshold lower or at payment threshold. DisconnectThreshold: %d, PaymentThreshold: %d", params.DisconnectThreshold, params.PaymentThreshold)
+		return nil, fmt.Errorf("disconnect threshold lower or at payment threshold. DisconnectThreshold: %d, PaymentThreshold: %d", params.DisconnectThreshold, params.PaymentThreshold)
 	}
+	// connect to the backend
 	backend, err := ethclient.Dial(backendURL)
 	if err != nil {
-		return nil, fmt.Errorf("swap init error: error connecting to Ethereum API %s: %s", backendURL, err)
+		return nil, fmt.Errorf("error connecting to Ethereum API %s: %v", backendURL, err)
 	}
-
-	// we may not need this check, and we could maybe even get rid of this constant completely
-	if params != nil && params.InitialDepositAmount == 0 {
-		// need to prompt user for initial deposit amount
-		// if 0, can not cash in cheques
-		prompter := console.Stdin
-
-		// ask user for input
-		input, err := prompter.PromptInput("Please provide the amount in Wei which will deposited to your chequebook upon deployment: ")
-		if err != nil {
-			return nil, err
-		}
-		// check input
-		val, err := strconv.ParseInt(input, 10, 64)
-		if err != nil {
-			// maybe we should provide a fallback here? A bad input results in stopping the boot
-			return nil, fmt.Errorf("Conversion error while reading user input: %v", err)
-		}
-		log.Info("Chequebook initial deposit amount: ", "amount", val)
-		params.InitialDepositAmount = uint64(val)
+	// get the networkID of the backend
+	var networkID *big.Int
+	if networkID, err = backend.NetworkID(context.TODO()); err != nil {
+		return nil, fmt.Errorf("error retrieving networkID from backendURL: %v", err)
 	}
-
-	swap := new(
+	// verify that we have not used SWAP before on a different networkID
+	if err := checkNetworkID(networkID.Uint64(), stateStore); err != nil {
+		return nil, err
+	}
+	auditLog.Info("SWAP initialized on backend network ID", "ID", networkID.Uint64())
+	// create the owner of SWAP
+	owner := createOwner(prvkey)
+	// start the chequebook
+	var contract contract.Contract
+	if contract, err = StartChequebook(chequebookAddressFlag, initialDepositAmountFlag, stateStore, owner, backend); err != nil {
```

Comment:
To be honest I don't like this change. I understand that we want to do this during the `New` function but the fact that `StartChequebook` (and `Deploy` etc.) are no longer defined on `Swap` leads to a lot of unnecessary argument passing which means that for example if we need to add new fields that effect deployment (e.g. the factory address) we need to change many function signatures and adjust many tests.

If we need to call `StartChequebook` here why don't we call `new` first and then call `StartChequebook` on that?

Reply:
One of the reason to introduce `Params` was just that - without a clear knowledge of which parameters were coming.

We can't stick everything there either though.

It slipped out of my attention that `StartChequebook` and `Deploy` are not on `Swap` anymore (great catch @ralph-pichler ), Any reason for that @Eknir ?

Reply:
I put the receivers back, changed a bit of the order in `swap.New`, now first calling `swap.new`, after which I call `swap.StartChequebook`